oneargopy/Argo.py

```diff
@@ -18,56 +18,296 @@
+
+            NOTE: If the user has their own settings configuration and has
+            set keep_index_in_memory to false then the dataframes will be
+            removed from memory at the end of construction and will be
+            reloaded with following Argo function calls, meaning that
+            functions will take longer but occupy less memory if this
+            option is set to false. 
         """
         self.download_settings = DownloadSettings(user_settings)
         self.source_settings = SourceSettings(user_settings)
-        if self.download_settings.verbose: print(f'Starting initialize process...\n')
+        if self.download_settings.verbose: print(f'Starting initialize process...')
         
         if self.download_settings.verbose: print(f'Your current download settings are: {self.download_settings}')
         if self.download_settings.verbose: print(f'Your current source settings are: {self.source_settings}')
 
         # Check for and create subdirectories if needed
-        if self.download_settings.verbose: print(f'Checking for subdirectories...\n')
+        if self.download_settings.verbose: print(f'Checking for subdirectories...')
         self.__initialize_subdirectories()
 
         # Download files from GDAC to Index directory
-        if self.download_settings.verbose: print(f'\nDownloading index files...\n')
+        if self.download_settings.verbose: print(f'\nDownloading index files...')
         for file in self.download_settings.index_files:
             self.__download_index_file(file)
 
-        # Load the argo_synthetic-profile_index.txt file into a dataframe
+        # Load the index files into dataframes
         if self.download_settings.verbose: print(f'\nTransferring index files into dataframes...')
-        self.sprof_index  = self.__load_sprof_dataframe()
+        self.sprof_index  = self.__load_sprof_dataframe() 
         self.prof_index = self.__load_prof_dataframe()
 
+        # Add column noting if a float profile is also in the sprof_index, meaning that it is a bgc float
+        if self.download_settings.verbose: print(f'Marking bgc floats in prof_index dataframe...')
+        self.__mark_bgcs_in_prof()
+
+        # Create float_is_bgc refrence index for use in select profiles
+        if self.download_settings.verbose: print(f'Creating float_is_bgc_index dataframe...')
+        self.float_is_bgc_index = self.__load_is_bgc_index()
+        
         # Print number of floats
         if self.download_settings.verbose: self.__display_floats() 
 
-        if self.download_settings.verbose: print(f'Initialize is finished!')
+        if self.download_settings.verbose: print(f'Initialize is finished!\n\n')
+
+        if not self.download_settings.keep_index_in_memory:
+            if self.download_settings.verbose: print('Removing dataframes from memory...')
+            del self.sprof_index 
+            del self.prof_index
+            del self.float_is_bgc_index
```

Comment:
float_is_bgc_index is much smaller (~20000 rows), so we should always keep it in memory, saves us the del/reload